include/linker/sections.h

```diff
@@ -66,6 +66,7 @@
 #define IMX_BOOT_DATA	.boot_hdr.data
 #define IMX_BOOT_IVT	.boot_hdr.ivt
 #define IMX_BOOT_DCD	.boot_hdr.dcd_data
+#define STM32_PREINIT	.preinit
```

Comment:
If you need a change to this file, you need a specific commit and explain clearly why this is needed.
Also, it has to be generic, you can't put vendor specific code here.

Reply:
I can understand it should be generic, that thing is the same problem was solved for the IMX boards by adding this vendor specific code in this way. I wasn't sure if I should name it with STM32...I guess not. We need a generic way of adding preinit code that sets up external memory. That's the whole reason this was done with preinit, so it could use the OCRAM (on-chip) for the stack whilst it sets up the FMC.

Reply:
Ok, then, can you try to make it as generic as possible, split in commits and provide more explanations in git messages?
Whole point is to help reviewers understanding your work, your choices, what it adds and how to use it.
As an example IMX code was added here: https://github.com/zephyrproject-rtos/zephyr/pull/12237